src/apps/api/views/leaderboards.py

```diff
@@ -23,9 +24,23 @@ def get_queryset(self):
         return qs
 
 
+class SubmissionScoreViewSet(ModelViewSet):
+    queryset = SubmissionScore.objects.all()
+    serializer_class = SubmissionScoreSerializer
+
+    def update(self, request, *args, **kwargs):
+        instance = self.get_object()
+        comp = instance.submission.phase.competition
+        admin_users = [comp.created_by] + list(comp.collaborators.all())
+        if request.user not in admin_users and not request.user.is_superuser:
```

Comment:
Should we maybe consider doing some kind of Permissions class if we end up using this kind of check often? Can make a separate issue for this!! really good candidate for a one-off refactor thing to track

Reply:
- [x] Issue created